agent-install/agent-install.sh

```diff
@@ -1441,31 +1550,53 @@ function check_existing_exch_node_is_correct_type() {
 
 # Cluster only: to extract agent image tar.gz and load to docker
 # Side-effect: sets globals: AGENT_IMAGE, AGENT_IMAGE_VERSION_IN_TAR, IMAGE_FULL_PATH_ON_EDGE_CLUSTER_REGISTRY
-function getImageInfo() {
-    log_debug "getImageInfo() begin"
+function loadAgentImage() {
+    log_debug "loadAgentImage() begin"
 
-    log_info "Unpacking amd64_anax_k8s_ubi.tar.gz ..."
-    tar xzf amd64_anax_k8s_ubi.tar.gz
-    chk $? 'uncompressing amd64_anax_k8s_ubi.tar.gz'
+    # Get the agent tar file, if necessary
+    if using_remote_input_files 'pkg'; then
+        if [[ -n $AGENT_IMAGE_TAR_FILE && $AGENT_IMAGE_TAR_FILE != $DEFAULT_AGENT_IMAGE_TAR_FILE ]]; then
+            log_fatal 1 "Can not specify both AGENT_IMAGE_TAR_FILE and -i (INPUT_FILE_PATH)"
+        fi
+        if [[ $INPUT_FILE_PATH == css:* ]]; then
+            download_css_file "$INPUT_FILE_PATH/$AGENT_IMAGE_TAR_FILE"
+        elif [[ $INPUT_FILE_PATH == https://github.com/open-horizon/anax/releases* ]]; then
+            #download_anax_release_file "$INPUT_FILE_PATH/$AGENT_IMAGE_TAR_FILE"
+            # Get the docker image from docker hub in this case
+            #todo: not sure how to make the user interface cleaner for this case, and to get the version we should pull
+            local image_path='openhorizon/amd64_anax_k8s:latest'
+            log_info "Pulling $image_path from docker hub..."
+            docker pull "$image_path"
+            chk $? "pulling $image_path"
+            AGENT_IMAGE=$image_path
+            AGENT_IMAGE_VERSION_IN_TAR=${AGENT_IMAGE##*:}
+            IMAGE_FULL_PATH_ON_EDGE_CLUSTER_REGISTRY="$IMAGE_ON_EDGE_CLUSTER_REGISTRY:$AGENT_IMAGE_VERSION_IN_TAR"
+            return
+        fi
+    fi
+
+    log_info "Unpacking $AGENT_IMAGE_TAR_FILE ..."
+    gunzip $AGENT_IMAGE_TAR_FILE   # this also removes the .gz from the file name
+    chk $? "uncompressing $AGENT_IMAGE_TAR_FILE"
 
-    # Loaded image: {repo}/{image_name}:{version_number}
-    local loaded_image_message=$(docker load --input amd64_anax_k8s_ubi.tar)
-    chk $? 'docker loading amd64_anax_k8s_ubi.tar'
+    local loaded_image_message=$(docker load --input ${AGENT_IMAGE_TAR_FILE%.gz})
```

Comment:
should this value have .gz? `AGENT_IMAGE_TAR_FILE%`

Reply:
It doesn't have `.gz` at the end. The syntax `${AGENT_IMAGE_TAR_FILE%.gz}` means remove `.gz` from the end of `AGENT_IMAGE_TAR_FILE`